library/core/src/ffi/c_str.rs

```diff
@@ -341,7 +342,8 @@ impl CStr {
         }
     }
 
-    /// Creates a C string wrapper from a byte slice.
+    /// Creates a C string wrapper from a byte slice terminated by exactly one
+    /// nul byte.
```

Comment:
Same comment on length, though if it really does need to be this long that's okay.